src/common/bottom_up_merge.cc

```diff
@@ -37,6 +37,106 @@ BottomUpMerge::BottomUpMerge(std::vector<std::unique_ptr<Dataset>>& datasets,
   }
 }
 
+//Add by akash
+BottomUpMerge::BottomUpMerge(nlohmann::json dataset_json_obj, 
+			     std::vector<std::unique_ptr<Dataset>>& datasets_old,
+			     std::vector<std::unique_ptr<Dataset>>& datasets,
+                             ProteinAligner* aligner) {
+  
+  // Add by akash : create a map from the of dataset and use that to store into old cluster set data structure
```

Comment:
keep comments relevant to code only. Your contributions/changes are recorded by git, that's what it's for ;-) 
only exceptions would the odd TODO item